gtests/freebl_gtest/cmac_unittests.cc

```diff
@@ -0,0 +1,225 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "gtest/gtest.h"
+
+#include <stdint.h>
+#include <memory>
+
+#include "blapi.h"
+#include "secitem.h"
+
+class CMACTest : public ::testing::Test {
+    protected:
+        CMACContext *CreateWithRandomKey(int bytes) {
+            unsigned char *key = PORT_NewArray(unsigned char, bytes);
+            return CMAC_AESCreate(key, bytes);
+        }
+
+        CMACContext *CreateWithKey(const unsigned char *key, int bytes) {
+            return CMAC_AESCreate(key, bytes);
+        }
+
+        bool Compare(const unsigned char *actual, const unsigned char *expected, unsigned int length) {
+            return strncmp((const char *)actual, (const char *)expected, length) == 0;
+        }
+};
+
+TEST_F(CMACTest, invalid_size) {
+    CMACContext *ctx = CreateWithRandomKey(1);
+    ASSERT_TRUE(ctx == nullptr);
+}
+
+TEST_F(CMACTest, right_size) {
+    CMACContext *ctx = CreateWithRandomKey(AES_128_KEY_LENGTH);
+    ASSERT_TRUE(ctx != nullptr);
+    CMAC_Destroy(ctx, PR_TRUE);
+}
+
+/* The following tests were taken from NIST's Cryptographic Standards and
```

Comment:
I explicitly wanted to break these out into separate tests instead of putting them in a data structure and looping.